app/main.rs

```diff
@@ -540,11 +540,26 @@ async fn main() -> Result<()> {
             )
         })?;
 
-        if !wallet.is_initialized().await && cli.wallet_opts.auto_create {
-            tracing::info!("auto-creating new wallet");
-            let mnemonic = None;
-            let password = None;
-            wallet.create_wallet(mnemonic, password).await?;
+        if !wallet.is_initialized().await {
+            if cli.wallet_opts.auto_create && cli.wallet_opts.mnemonic_path.is_some() {
+                return Err(miette!(
+                    "Cannot use both --wallet-auto-create and --set-seed-with-mnemonic"
+                ));
+            }
```

Comment:
This can be done with the `conflicts_with` attribute in the config struct instead, with the `conflicts_with` attribute